src/EditorFeatures/Core/Implementation/CallHierarchy/Finders/AbstractCallFinder.cs

```diff
@@ -48,51 +50,64 @@ public void CancelSearch()
             _cancellationSource.Cancel();
         }
 
-        public void StartSearch(CallHierarchySearchScope searchScope, ICallHierarchySearchCallback callback)
+        public void StartSearch(Workspace workspace, CallHierarchySearchScope searchScope, ICallHierarchySearchCallback callback)
         {
-            Task.Run(() => SearchAsync(callback, searchScope, _cancellationSource.Token), _cancellationSource.Token);
-        }
-
-        private async Task SearchAsync(ICallHierarchySearchCallback callback, CallHierarchySearchScope scope, CancellationToken cancellationToken)
-        {
-            callback.ReportProgress(0, 1);
-
             var asyncToken = _asyncListener.BeginAsyncOperation(this.GetType().Name + ".Search");
 
-            // Follow the search task with task that lets the callback know we're done and which
-            // marks the async operation as complete.  Note that we pass CancellationToken.None
-            // here.  That's intentional.  This operation is *not* cancellable.
+            // NOTE: This task has CancellationToken.None specified, since it must complete no matter what
+            // so the callback is appropriately notified that the search has terminated.
+            Task.Run(async () =>
+            {
+                // The error message to show if we had an error. null will mean we succeeded.
+                string completionErrorMessage = null;
+                try
+                {
+                    await SearchAsync(workspace, searchScope, callback, _cancellationSource.Token).ConfigureAwait(false);
+                }
+                catch (OperationCanceledException)
+                {
+                    completionErrorMessage = EditorFeaturesResources.Canceled;
+                }
+                catch (Exception e) when (FatalError.ReportWithoutCrash(e))
+                {
+                    completionErrorMessage = e.Message;
```

Comment:
So we're going to tell the user "null reference exception" ? :)

Reply:
I'm OK with that. :smile:

Reply:
(more seriously: I could give a "something broke" message but not sure that actually helps anyone...)